src/ethereum_test_exceptions/exception_mapper.py

```diff
@@ -0,0 +1,57 @@
+"""
+EEST Exception mapper
+"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from bidict import frozenbidict
+
+from .exceptions import ExceptionBase, UndefinedException
+
+
+@dataclass
+class ExceptionMessage:
+    """Defines a mapping between an exception and a message."""
+
+    exception: ExceptionBase
+    message: str
+
+
+class ExceptionMapper(ABC):
+    """
+    Translate between EEST exceptions and error strings returned by client's t8n or other tools.
+    """
+
+    def __init__(self) -> None:
+        # Ensure that the subclass has properly defined _mapping_data before accessing it
+        assert self._mapping_data is not None, "_mapping_data must be defined in subclass"
+
+        assert len(set(entry.exception for entry in self._mapping_data)) == len(
+            self._mapping_data
+        ), "Duplicate exception in _mapping_data"
+        assert len(set(entry.message for entry in self._mapping_data)) == len(
+            self._mapping_data
+        ), "Duplicate message in _mapping_data"
+        self.exception_to_message_map: frozenbidict = frozenbidict(
+            {entry.exception: entry.message for entry in self._mapping_data}
+        )
+
+    @property
+    @abstractmethod
+    def _mapping_data(self):
+        """This method should be overridden in the subclass to provide mapping data."""
+        pass
+
+    def exception_to_message(self, exception: ExceptionBase) -> str:
+        """Takes an exception and returns a formatted string."""
+        message = self.exception_to_message_map.get(exception, "Undefined")
```

Comment:
```suggestion
    def exception_to_message(self, exception: ExceptionBase) -> str | None:
        """Takes an exception and returns a formatted string."""
        message = self.exception_to_message_map.get(exception, None)
```
I think we should return `None` here and then handle this case in `verify_transaction_exception`, because with `"Undefined"` the exception we get on failure is:
```
E               Exception: TransactionException (pos=1, nonce=0x5)
E               Error: exception mismatch:
E                want = 'Undefined' (TransactionException.PRIORITY_GREATER_THAN_MAX_FEE_PER_GAS),
E                got  = 'Failed transaction: ' (TransactionException.TYPE_4_TX_CONTRACT_CREATION)
E                No message defined for TransactionException.PRIORITY_GREATER_THAN_MAX_FEE_PER_GAS, please add it to ExecutionSpecsExceptionMapper
```
which is confusing.

Reply:
this is actually not confusing. it tells me that ExecutionSpecsExceptionMapper is missing string for `TransactionException.PRIORITY_GREATER_THAN_MAX_FEE_PER_GAS`

Reply:
What I meant is that it's looking for the string `'Undefined'` in the error message returned by the transition tool, which I think should not be the case. It should stop processing before at a prior point in the code because it's never going to find `'Undefined'` in the message (or maybe it will and that's a bit worse because it's a false positive).

Reply:
I see, I added the None return and replaced checks against "Undefined" string to checks against None